src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,109 @@
-import React from 'react';
-import { NewCommentForm } from '../NewCommentForm';
+import React, { useCallback, useEffect, useState } from 'react';
+import { addComment, deleteComment, getPostComments } from '../../api/comments';
+import { getPostDetails } from '../../api/posts';
+import { Post } from '../../types/Post';
+import { Comment, FetchComment } from '../../types/Comment';
 import './PostDetails.scss';
+import { NewCommentForm } from '../NewCommentForm';
+
+interface Props {
+  selectedPostId: number
+}
+
+export const PostDetails: React.FC<Props> = React.memo(({ selectedPostId }) => {
+  const [details, setDetails] = useState<Post>();
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [showComments, setShowComments] = useState(true);
+
+  const loadBody = (postId: number) => {
+    getPostDetails(postId)
+      .then(setDetails);
+  };
+
+  const loadComments = (postId: number) => {
+    getPostComments(postId)
+      .then(setComments);
+  };
+
+  const removeComment = (commentId: number) => {
+    setComments(comments.filter(comment => comment.id !== commentId));
+    deleteComment(commentId);
+  };
+
+  const createComment = useCallback((newComment: FetchComment) => {
+    const commentToList = {
+      ...newComment,
+      id: Date.now().valueOf(),
+    };
+
+    setComments(currentComments => [...currentComments, commentToList]);
+    addComment(newComment);
```

Comment:
await a response from the server and then add comment (which returned query) to state comment
https://www.loom.com/share/66115c6ef2bc4a079424b1bef021c669